x-pack/plugins/lens/public/embeddable/embeddable.tsx

```diff
@@ -417,6 +421,153 @@ export class Embeddable
     );
   }
 
+  private get activeDatasourceId() {
+    return getActiveDatasourceIdFromDoc(this.savedVis);
```

Comment:
nit: maybe I'm over-engineering here, but can we memoize the returning value of this?
I see it's called multiple times later on and it's producing and iterate to pick the first value every time.
Something like:
```
if(!this._activeDatasourceId){
  this._activeDatasourceId = getActiveDatasourceIdFromDoc(this.savedVis);
}
return this._activeDatasourceId;
```

Reply:
I personally I don't see this as an expensive-enough operation to merit caching. True, it iterates over the datasource states, but that is max two objects. However, I'm willing to do this if you still feel like we should! Just let me know.

Reply:
I know, that's why it is a bit over-engineered. No problem

Reply:
Eh, never mind. Can't hurt to do it 🤷